### PR TITLE
chore(hybrid-cloud): Simplify Discord API client

### PR DESCRIPTION
**Pull request by dashed**
The Discord API client has no business being an integration proxy API client since it doesn't rely on any token refresh dance, nor any dependence on any integration database records on the Control silo.

This refactor lets the Discord API client to make Discord API calls from within the Region silo itself, rather than performing it at the control silo.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -372,6 +372,7 @@ module = [
     "sentry.integrations.bitbucket_server.client",
     "sentry.integrations.bitbucket_server.integration",
     "sentry.integrations.bitbucket_server.webhook",
+    "sentry.integrations.discord.utils.channel",
```
Comment (markstory) [inline pyproject.toml:375]: We should be avoiding adding to this list.

File: src/sentry/integrations/discord/client.py [hunk 2]
```diff
@@ -62,54 +57,19 @@ def get_guild_name(self, guild_id: str) -> str:
         response = self.get(GUILD_URL.format(guild_id=guild_id), headers=self.prepare_auth_header())
         return response["name"]  # type: ignore
 
-
-class DiscordClient(IntegrationProxyClient):
```
Comment (markstory) [inline src/sentry/integrations/discord/client.py:66]: Removing the integration proxy will help with reliability and operations, but comes with an increased cognitive cost of remembering which integrations need to use the proxy and which don't and why.  Keeping that in order as we add more integrations could be challenging for future us 🤔

**Review by markstory — APPROVED** [review]
Looks good outside of the typing issue that should be fixed before merge.

File: pyproject.toml [hunk 3]
```diff
@@ -372,6 +376,7 @@ module = [
...
```
Comment (asottile-sentry) [inline pyproject.toml:379]: the allowlist should not be increased -- we're trying to _improve_ type checking not regress it

File: pyproject.toml [hunk 4]
```diff
@@ -56,7 +56,11 @@ looponfailroots = ["src", "tests"]
 [tool.mypy]
 python_version = "3.8"
 mypy_path = ["fixtures/stubs-for-mypy"]
-plugins = ["pydantic.mypy", "mypy_django_plugin.main", "tools.mypy_helpers.plugin"]
+plugins = [
+    "pydantic.mypy",
+    "mypy_django_plugin.main",
+    "tools.mypy_helpers.plugin",
+]
```
Comment (asottile-sentry) [inline pyproject.toml:63]: unrelated formatting change

**Review by asottile-sentry — DISMISSED** [review]

File: pyproject.toml (hunk 3 above)
Comment (dashed) [reply pyproject.toml:379]: @asottile-sentry i'm trying to remove it. I'm unsure why it's adding it. 

`mypy` passes locally for me.

Comment (asottile-sentry) [reply pyproject.toml:379]: lemme check it out and see what's going on here...

Comment (dashed) [reply pyproject.toml]: @asottile-sentry I think fix might be https://github.com/getsentry/sentry/pull/61655/commits/856cec1e7d308a466976fe4354d6ea7a297efe78. I'm unsure why `mypy` didn't complain for me locally 🤔

Comment (asottile-sentry) [reply pyproject.toml]: ah I found it --

```
$ grep utils/channel .artifacts/mypy-all
src/sentry/integrations/slack/utils/channel.py:133: error: Argument 2 to "check_for_channel" has incompatible type "Optional[str]"; expected "str"  [arg-type]
src/sentry/integrations/slack/utils/channel.py:202: error: Unsupported right operand type for in ("Union[BaseApiResponse, Mapping[str, Any], Response]")  [operator]
src/sentry/integrations/slack/utils/channel.py:207: error: Value of type "Union[BaseApiResponse, Mapping[str, Any], Response]" is not indexable  [index]
src/sentry/integrations/slack/utils/channel.py:208: error: Value of type "Union[BaseApiResponse, Mapping[str, Any], Response]" is not indexable  [index]
src/sentry/integrations/slack/utils/channel.py:213: error: Value of type "Union[BaseApiResponse, Mapping[str, Any], Response]" is not indexable  [index]
src/sentry/integrations/discord/utils/channel.py:38: note: "validate_channel_id" defined here
```

the allowlist-thingy added `utils.channel` due to the `note:` -- the error was actually coming from a different error though! (which is allowlisted at the moment)

Comment (dashed) [reply pyproject.toml]: Oh interesting. Thanks for checking!
